### PR TITLE
Consider initContainer GPUs quota in calculating

### PR DESCRIPTION
**Pull request by kerthcet**
Signed-off-by: kerthcet <kerthcet@gmail.com>

fix: https://github.com/volcano-sh/volcano/issues/2422

### CODE REVIEW

File: pkg/scheduler/api/device_info.go [hunk 1]
```diff
@@ -55,15 +55,27 @@ func (g *GPUDevice) getUsedGPUMemory() uint {
...
+
+	if mem > initMem {
+		return mem
+	}
+	return initMem
 }
 
 // getGPUResourceOfPod returns the GPU resource required by the container.
-func getGPUResourceOfContainer(container *v1.Container) uint {
+func getGPUResourceOfContainer(container v1.Container) uint {
```
Comment (Thor-wl) [inline pkg/scheduler/api/device_info.go:78]: Suggest to keep the object reference as the input parameter for we only need the `spec.resource` part. That will be more efficient.

Comment (kerthcet) [reply pkg/scheduler/api/device_info.go:78]: Make sense, currently we pass a value here.

**Comment by kerthcet** [comment]
Updated and squashed.
The flaky test maybe not related to this PR.
```
failed to create job queue2-job in namespace 
  Unexpected error:
      <*errors.StatusError | 0xc00044e960>: {
          ErrStatus: {
              TypeMeta: {Kind: "", APIVersion: ""},
              ListMeta: {
                  SelfLink: "",
                  ResourceVersion: "",
                  Continue: "",
                  RemainingItemCount: nil,
              },
              Status: "Failure",
              Message: "admission webhook \"validatejob.volcano.sh\" denied the request: can only submit job to queue with state `Open`, queue `queue2` status is ``;",
              Reason: "",
              Details: nil,
              Code: 400,
          },
      }
      admission webhook "validatejob.volcano.sh" denied the request: can only submit job to queue with state `Open`, queue `queue2` status is ``;
  occurred
```

**Review by Thor-wl — APPROVED** [review]

**Review by hwdef — COMMENTED** [review]
/lgtm
Thanks!

**Comment by Thor-wl** [comment]
@kerthcet Hi, Could you cherry pick this patch to branch v1.6/v.15/v1.4? It makes sense.

**Comment by kerthcet** [comment]
I think you mean release-1.6, release-1.5, release-1.4, right?

**Comment by Thor-wl** [comment]
> I think you mean release-1.6, release-1.5, release-1.4, right?

Yes.

**Pull request merged**